test/functional/rpc_fundrawtransaction.py

```diff
@@ -708,74 +707,108 @@ def test_all_watched_funds(self):
+
+        result = node.fundrawtransaction(rawtx)  # uses self.min_relay_tx_fee (set by settxfee)
+        result1 = node.fundrawtransaction(rawtx, {"fee_rate": 200000 * self.min_relay_tx_fee})
+        result2 = node.fundrawtransaction(rawtx, {"feeRate": 2 * self.min_relay_tx_fee})
+        result3 = node.fundrawtransaction(rawtx, {"fee_rate": 1000000 * self.min_relay_tx_fee})
+        result4 = node.fundrawtransaction(rawtx, {"feeRate": 10 * self.min_relay_tx_fee})
         result_fee_rate = result['fee'] * 1000 / count_bytes(result['hex'])
+        assert_fee_amount(result1['fee'], count_bytes(result2['hex']), 2 * result_fee_rate)
         assert_fee_amount(result2['fee'], count_bytes(result2['hex']), 2 * result_fee_rate)
         assert_fee_amount(result3['fee'], count_bytes(result3['hex']), 10 * result_fee_rate)
+        assert_fee_amount(result4['fee'], count_bytes(result3['hex']), 10 * result_fee_rate)
 
-    def test_feerate_with_conf_target_and_estimate_mode(self):
-        self.log.info("Test fundrawtxn passing an explicit fee rate using conf_target and estimate_mode")
-        node = self.nodes[3]
-        # Make sure there is exactly one input so coin selection can't skew the result.
-        assert_equal(len(node.listunspent(1)), 1)
-        inputs = []
-        outputs = {node.getnewaddress() : 1}
-        rawtx = node.createrawtransaction(inputs, outputs)
-
-        for unit, fee_rate in {"btc/kb": 0.1, "sat/b": 10000}.items():
-            self.log.info("Test fundrawtxn with conf_target {} estimate_mode {} produces expected fee".format(fee_rate, unit))
-            # With no arguments passed, expect fee of 141 sats/b.
-            assert_approx(node.fundrawtransaction(rawtx)["fee"], vexp=0.00000141, vspan=0.00000001)
-            # Expect fee to be 10,000x higher when explicit fee 10,000x greater is specified.
-            result = node.fundrawtransaction(rawtx, {"conf_target": fee_rate, "estimate_mode": unit})
-            assert_approx(result["fee"], vexp=0.0141, vspan=0.0001)
+        msg = "Fee exceeds maximum configured by user (e.g. -maxtxfee, maxfeerate)"
+        assert_raises_rpc_error(-4, msg, node.fundrawtransaction, rawtx, {"fee_rate": 100000})
+        assert_raises_rpc_error(-4, msg, node.fundrawtransaction, rawtx, {"feeRate": 1})
 
-        for field, fee_rate in {"conf_target": 0.1, "estimate_mode": "sat/b"}.items():
-            self.log.info("Test fundrawtxn raises RPC error if both feeRate and {} are passed".format(field))
-            assert_raises_rpc_error(
-                -8, "Cannot specify both {} and feeRate".format(field),
-                lambda: node.fundrawtransaction(rawtx, {"feeRate": 0.1, field: fee_rate}))
+        # With no arguments passed, expect fee of 141 sats/b.
+        assert_approx(node.fundrawtransaction(rawtx)["fee"], vexp=0.00000141, vspan=0.00000001)
+        # Expect fee to be 10,000x higher when explicit fee 10,000x greater is specified.
```

Comment:
This would in turn mean that this sentence should be:

 > Expect fee to be 10,000x higher when explicit **feerate** 10,000x greater is specified.

Also, nit: Wouldn't "10,000× higher" be "10,001 as high"? :nerd_face:

Reply:
Well-spotted :)
```diff
-        # With no arguments passed, expect fee of 141 sats/b.
+        # With no arguments passed, expect fee of 141 satoshis.
         assert_approx(node.fundrawtransaction(rawtx)["fee"], vexp=0.00000141, vspan=0.00000001)
-        # Expect fee to be 10,000x higher when explicit fee 10,000x greater is specified.
+        # Expect fee to be 10,000x higher when explicit fee rate 10,000x greater is specified.
```